pkg/operator/controller/controller.go

```diff
@@ -95,10 +101,24 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		ingress = nil
 	}
 
-	caSecret, err := r.ensureRouterCACertificateSecret()
-	if err != nil {
-		errs = append(errs, fmt.Errorf("failed to ensure CA secret: %v", err))
-	} else {
+	caSecretName := cacert.CASecretName(r.Namespace)
+	caSecret := &corev1.Secret{}
+	if err := r.Client.Get(context.TODO(), caSecretName, caSecret); err != nil {
+		if errors.IsNotFound(err) {
+			// Most likely the CA cert controller hasn't created it yet, try again
+			// after a reasonable time.
+			result.RequeueAfter = 5 * time.Second
```

Comment:
I thought there were problems using `currentRouterCASecret`, so we were no longer using it (see #132).

Reply:
Error take precedence over requeueAfter when it comes to the result processing, which makes sense — if there's no error, requeueAfter would be effective. Does that seem okay here?

Reply:
Yeah, seems reasonable.